src/wallet/did_wallet/did_wallet.py

```diff
@@ -0,0 +1,760 @@
+        signature = AugSchemeMPL.sign(private, message)
+        # assert signature.validate([signature.PkMessagePair(pubkey, message)])
+        sigs = [signature]
+        aggsig = AugSchemeMPL.aggregate(sigs)
+        spend_bundle = SpendBundle(list_of_solutions, aggsig)
+
+        did_record = TransactionRecord(
+            confirmed_at_index=uint32(0),
+            created_at_time=uint64(int(time.time())),
+            to_puzzle_hash=puzhash,
+            amount=uint64(coin.amount),
+            fee_amount=uint64(0),
+            incoming=False,
+            confirmed=False,
+            sent=uint32(0),
+            spend_bundle=spend_bundle,
+            additions=spend_bundle.additions(),
+            removals=spend_bundle.removals(),
+            wallet_id=self.wallet_info.id,
+            sent_to=[],
+            trade_id=None,
+        )
+        await self.standard_wallet.push_transaction(did_record)
+        return spend_bundle
+
+    async def create_attestment(self, identity, newpuz):
+        coins = await self.select_coins(1)
+        coin = coins.pop()
+        message = did_wallet_puzzles.get_recovery_message_puzzle(identity, newpuz)
+        innermessage = message.get_tree_hash()
+        # innerpuz solution is (mode amount new_puz identity my_puz)
+        innersol = Program.to(
+            [1, coin.amount, innermessage, identity, coin.puzzle_hash]
+        )
+        # full solution is (corehash parent_info my_amount innerpuz_reveal solution)
+        innerpuz = self.did_info.current_inner
+        full_puzzle: str = did_wallet_puzzles.create_fullpuz(
+            innerpuz,
+            self.did_info.my_did,
+        )
+        parent_info = await self.get_parent_for_coin(coin)
+
+        fullsol = Program.to(
+            [
+                [
+                    parent_info.parent_name,
+                    parent_info.inner_puzzle_hash,
+                    parent_info.amount,
+                ],
+                coin.amount,
+                innersol,
+            ]
+        )
+        list_of_solutions = [
+            CoinSolution(
+                coin,
+                clvm.to_sexp_f([full_puzzle, fullsol]),
+            )
+        ]
+        message_spend = did_wallet_puzzles.create_spend_for_message(
+            coin.name(), identity, newpuz
+        )
+
+        message_spend_bundle = SpendBundle([message_spend], AugSchemeMPL.aggregate([]))
+        # sign for AGG_SIG_ME
+        message = bytes(innermessage) + bytes(coin.name())
+        pubkey = did_wallet_puzzles.get_pubkey_from_innerpuz(innerpuz)
+        index = await self.wallet_state_manager.puzzle_store.index_for_pubkey(pubkey)
+        private = master_sk_to_wallet_sk(self.wallet_state_manager.private_key, index)
+        signature = AugSchemeMPL.sign(private, message)
+        # assert signature.validate([signature.PkMessagePair(pubkey, message)])
+        sigs = [signature]
+        aggsig = AugSchemeMPL.aggregate(sigs)
```

Comment:
If there is only one signature, as there is here, this is a no-op, ie. you don't need to do this.